src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -306,3 +311,214 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
+        test_harness.writeFadRecords(sas_test_harness_dump_records)
+
+      # Step 14,15 : Trigger Full Activty Dump and Pull FAD records from SAS UUT
+      fad_uut_object = getFullActivityDumpSasUut(self._sas,
+                                                 self._sas_admin,
+                                                 sas_uut_cert,
+                                                 sas_uut_key)
+
+      # Step 16 : Trigger CPAS and wait for its completion
+      self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+      # Step 17 : Call IAP reference model
+      # TODO: To invoke DPA Move Reference model
+
+      # Call Pre IAP model
+      # TODO: To invoke Pre-iap reference model
+
+      # Call IAP reference model
+      # TODO: To invoke IAP reference model
+
+      # Step 18,19,20 and 21 :
+      # Send heartbeat request for the grants, relinquish the grant, grant request and heartbeat for new grant
+      for domain_proxy_object in domain_proxy_objects:
+        domain_proxy_object.performHeartbeatAndUpdateGrants()
+
+        # Step 22: Calculating the Aggregate interface and IAP reference model invoke.
+        cbsd_agg_list.append(domain_proxy_object.getCbsdsWithAtLeastOneAuthorizedGrant())
+
+      # Invoke Aggregate Interference Model
+      # TODO: To invoke Aggregate Interference Model
+
+      # Check MCP.1
+      # TODO: To invoke checkMcpIap method
+
+      # Execute from Step23 to Step30 only for MCP testcase
+      if test_type == "MCP":
+        # Step 23: ESC Test harness enables DPA activations
+        # deactivated the dpaDeactivationList DPA IDs
+        for dpa in iteration_content['dpaActivationList']:
+          self._sas_admin.TriggerDpaDeactivation(dpa)
+
+        # activated the dpaActivationList DPA IDs
+        # step 24: wait for 240 sec if DPA is activated in step 23 else 15 sec
+        for dpa in iteration_content['dpaActivationList']:
+          self._sas_admin.TriggerDpaActivation(dpa)
+        if len(iteration_content['dpaActivationList']):
+          time.sleep(240)
+        else:
+          time.sleep(15)
+
+        # Step 25,26,27 and 28 :
+        # Send heartbeat request for the grants, relinquish the grant, grant request and heartbeat for new grant
+        for domain_proxy_object in domain_proxy_objects:
+          domain_proxy_object.performHeartbeatAndUpdateGrants()
+
+          # Step 29
+          # TODO: Aggregate interference for DPA
+
+          # Step 30:Invoke Aggregate Interference Model
+          cbsd_agg_list.append(domain_proxy_object.getCbsdsWithAtLeastOneAuthorizedGrant())
+
+        # Invoke Aggregate Interference Model
+        # TODO: To invoke Aggregate Interference Model
+
+        # Check MCP.1
+        # TODO: To invoke checkMcpIap method
+
+        # check MCP.1 DPA
+        # TODO: need to invoke the DPA checkperformHeartbeatAndUpdateGrants
+
+      for test_harness in sas_test_harness_objects:
```

Comment:
I think this is at the wrong level (big file). Currently it is run every iteration.